Makefile

```diff
@@ -0,0 +1,31 @@
+.PHONY: all build test gen-mocks verify-mocks
+
+GENERATED_PATH = pkg/transport/mocktransport/\*.go
+GO_SOURCES = $(shell find pkg -type f -name '*.go' ! -path $(GENERATED_PATH))
+GENERATED_SOURCES = $(shell find pkg -type f -name '*.go' -path $(GENERATED_PATH))
+PKGS = $(shell go list ./pkg/...)
+
+all: vendor verify-mocks build
+	@echo "To run the tests ensure kafka and zookeeper are running locally then issue 'make test'"
+
+vendor: glide.lock
+	glide install -v --force
+
+glide.lock: glide.yaml
+	glide up -v --force
+
+build: $(GO_SOURCES) vendor
```

Comment:
Will remove `vendor` from dependencies here, to be consistent with other projects.
The problem with this approach (while it first seemed sensible to me) is that git will "`touch`" this directory (and the glide files) at each checkout, thus triggering a rebuild. Given that we're putting the contents of vendor under source control, this is not needed / noisy. Updating `vendor` (ie `make vendor`) should be seen as a "manual" thing to do, in this approach

Reply:
As a matter of fact, had to update glide manifest and re-vendor everything to add test dependencies.

Reply:
@ericbottard Thanks for sorting this out.

Reply:
Actually, I think I don't make any sense.
function-controller has its build target not depend on `vendor`, while http-gateway does. And I think the latter scheme is fine, because git will "touch" all files that are updated by a commit.

If go sources are changed, make would rebuild w/o re-vendoring. Fine
If glide.* are changed, make would revendor, alterting the next developer that something was forgotten. Not ideal, but better than doing nothing.

I'll re-add those dependencies

Reply:
Err, http-gateway *doesn't*. Still. Let me run some tests.

Reply:
To clarify: the dockerize task *did* depend on vendor, the build task didn't, hence my confusion. After some testing, I think all of build, test, dockerize and debug-dockerize can depend on vendor/